config/config-api/test/com/thoughtworks/go/config/TemplatesConfigTest.java

```diff
@@ -135,7 +170,37 @@ public void shouldReturnFalseIfUserCannotEditTemplate() {
         PipelineTemplateConfig template = PipelineTemplateConfigMother.createTemplate(templateName, new Authorization(new AdminsConfig(new AdminUser(templateAdmin))),
                 StageConfigMother.manualStage("stage-name"));
         TemplatesConfig templates = new TemplatesConfig(template);
-        assertThat(templates.canUserEditTemplate(templateName, templateAdminWhoDoesNotHavePermissionToThisTemplate), is(false));
+        assertThat(templates.canUserEditTemplate(templateName, templateAdminWhoDoesNotHavePermissionToThisTemplate, null), is(false));
+    }
+
+    @Test
+    public void shouldReturnTrueIfUserWithARoleCanEditTemplate() {
+        CaseInsensitiveString templateAdmin = new CaseInsensitiveString("template-admin");
+        Role securityConfigRole = new Role(new CaseInsensitiveString("role1"), new RoleUser(templateAdmin));
+        AdminRole templateAdminRole = new AdminRole(securityConfigRole);
+        List<Role> roles = new ArrayList<>();
+        roles.add(securityConfigRole);
+        String templateName = "template1";
+        PipelineTemplateConfig template = PipelineTemplateConfigMother.createTemplate(templateName, new Authorization(new AdminsConfig(templateAdminRole)),
+                StageConfigMother.manualStage("stage-name"));
+        TemplatesConfig templates = new TemplatesConfig(template);
+
+        assertThat(templates.canUserEditTemplate(templateName, templateAdmin, roles), is(true));
```

Comment:
Similar to above. Maybe pull out methods like `roleWithUser("template-admin")` so that the whole bit around wrapping the `CaseInsensitiveString` in a `Role` and then in an `AdminRole` which is not very relevant to this test can go away and make this test easier to read.